src/ctapipe/tools/stats_calculation.py

```diff
@@ -0,0 +1,179 @@
+
+
+class StatisticsCalculatorTool(Tool):
+    """
+    Perform statistics calculation for DL1a image data
+    """
+
+    name = "StatisticsCalculatorTool"
+    description = "Perform statistics calculation for DL1a image data"
+
+    examples = """
+    To calculate statistics of DL1a image data files:
+
+    > ctapipe-stats-calculation --input_url input.dl1.h5 --output_path /path/monitoring.h5 --overwrite
+
+    """
+
+    input_url = Path(
+        help="Input CTA HDF5 files including DL1a image data",
+        allow_none=True,
+        exists=True,
+        directory_ok=False,
+        file_ok=True,
+    ).tag(config=True)
+
+    allowed_tels = Set(
+        trait=CInt(),
+        default_value=None,
+        allow_none=True,
+        help=(
+            "List of allowed tel_ids, others will be ignored. "
+            "If None, all telescopes in the input stream will be included."
+        ),
+    ).tag(config=True)
+
+    dl1a_column_name = CaselessStrEnum(
+        ["image", "peak_time", "variance"],
+        default_value="image",
+        allow_none=False,
+        help="Column name of the DL1a image data to calculate statistics",
+    ).tag(config=True)
+
+    output_column_name = Unicode(
+        default_value="statistics",
+        allow_none=False,
+        help="Column name of the output statistics",
+    ).tag(config=True)
+
+    output_path = Path(
+        help="Output filename", default_value=pathlib.Path("monitoring.h5")
+    ).tag(config=True)
+
+    overwrite = Bool(help="Overwrite output file if it exists").tag(config=True)
+
+    aliases = {
+        ("i", "input_url"): "StatisticsCalculatorTool.input_url",
+        ("o", "output_path"): "StatisticsCalculatorTool.output_path",
+    }
+
+    flags = {
+        "overwrite": (
+            {"StatisticsCalculatorTool": {"overwrite": True}},
+            "Overwrite existing files",
+        ),
+    }
+
+    classes = classes_with_traits(PixelStatisticsCalculator)
+
+    def setup(self):
+        # Check that the input and output files are not the same
+        if self.input_url == self.output_path:
+            raise ToolConfigurationError(
+                "Input and output files are same. Fix your configuration / cli arguments."
+            )
+
+        # Load the subarray description from the input file
+        subarray = SubarrayDescription.from_hdf(self.input_url)
+        # Initialization of the statistics calculator
+        self.stats_calculator = PixelStatisticsCalculator(
+            parent=self, subarray=subarray
+        )
+        # Read the input data with the 'TableLoader'
+        input_data = TableLoader(input_url=self.input_url)
+        # Get the telescope ids from the input data or use the allowed_tels configuration
+        tel_ids = subarray.tel_ids if self.allowed_tels is None else self.allowed_tels
+        # Read the whole dl1 images
+        self.dl1_tables = input_data.read_telescope_events_by_id(
```

Comment:
Why load all data for all telescopes into memory when you then loop over them one by one?